### PR TITLE
Update two remaining endpoints to get JSON RequestBody

### LINKED ISSUE #84: Some endpoints doesn't accept JSON payloads
**Is your feature request related to a problem? Please describe.**
Update these 2 endpoints to get a json requestbody.
1. Update mentor state
2. Approve or reject mentee

**Describe the solution you'd like**
Update the 2 endpoints and unit tests

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Piumal1999**
## Purpose

The purpose of this PR is to fix #84 

## Goals

Update two remaining endpoints to get JSON RequestBody in following APIs
1. Approve or reject mentee
2. Update state of mentor

## Approach

Added Map<> object as the Request Body
Updated the unit tests

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Related PRs
 
N/A

## Test environment
 
Ubuntu 19.04
OpenJDK 1.8

## Learning

Discussion:
https://sef.discourse.group/t/building-a-mock-api-server-for-the-scholarx-api/186/9

### CODE REVIEW

File: src/main/java/org/sefglobal/scholarx/controller/MenteeController.java [hunk 1]
```diff
@@ -24,12 +25,11 @@ public MenteeController(MenteeService menteeService) {
...
 
-
     @PutMapping("/{id}/state")
     @ResponseStatus(HttpStatus.OK)
     public Mentee approveOrRejectMentee(@PathVariable long id,
-                                        @Valid @RequestBody boolean isApproved)
+                                        @Valid @RequestBody Map<String, Boolean> payload)
             throws ResourceNotFoundException, BadRequestException {
-        return menteeService.approveOrRejectMentee(id, isApproved);
+        return menteeService.approveOrRejectMentee(id, payload.get("isApproved"));
```
Comment (YohanAvishke) [inline src/main/java/org/sefglobal/scholarx/controller/MenteeController.java:33]: Null check before sending. If `isApproved ` is not in the payload a type mismatch error will occure

**Review by YohanAvishke — CHANGES_REQUESTED** [review]

File: src/main/java/org/sefglobal/scholarx/controller/MenteeController.java (hunk 1 above)
Comment (Piumal1999) [reply src/main/java/org/sefglobal/scholarx/controller/MenteeController.java:37]: I added the null check in the controller itself. Or is it better if the whole payload is sent to the service method and the check is done in there ?

Comment (YohanAvishke) [reply src/main/java/org/sefglobal/scholarx/controller/MenteeController.java:37]: no it's ok to validate the payload in the controller

**Review by YohanAvishke — APPROVED** [review]
Great job @Piumal1999

**Pull request merged**